### ISSUE TITLE
Consider not allocating in the empty archetype on world.spawn()

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#1556]
> - ex: return something like EntityMutUninit, which turns into EntityMut after an insert or insert_bundle op
> - this actually regressed performance last time i tried it, but in theory it should be faster

From Future Work of #1525.

**Comment by james7132** [comment]
Is this still relevant after #6054? Seems like we're now avoiding this empty Archetype allocation on any non-trivial spawn.
